src/libraries/System.Runtime/ref/System.Runtime.csproj

```diff
@@ -13,4 +13,7 @@
   <ItemGroup>
     <Compile Include="System.Runtime.cs" />
   </ItemGroup>
+  <ItemGroup>
+    <Compile Include="$(CommonPath)System\Obsoletions.cs" Link="Common\System\Obsoletions.cs" />
```

Comment:
The `ref/System.Runtime.cs` file should use the full literal string, not these constants. The file is auto re-generated from time to time and the auto generation would convert the constants into literal strings anyway.

Reply:
@jkotas just to confirm, do you mean changing it to:

`<Compile Include="..\..\Common\src\System\Obsoletions.cs" Link="Common\System\Obsoletions.cs" />`? Or is there a better way to resolve the path than this?

Reply:
He meant the messages and diagnostic is strings. Instead of pulling in `Obsoletions.cs` at all. Look for `SYSLIB003` in the ref .cs file to see the existing ones. We copy/paste the literal strings into the ref files.

Reply:
Gotcha, will make that change, thanks

Reply:
Updated in https://github.com/dotnet/runtime/pull/39261/commits/29f28769f25628bf532a382f6d7f9e89419e5413

Reply:
> The file is auto re-generated from time to time and the auto generation would convert the constants into literal strings anyway.

Unrelated to this PR but I am curious, how / where does the process take place?

Reply:
> Unrelated to this PR but I am curious, how / where does the process take place?

The documentation is at https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/updating-ref-source.md. The re-generation is done manually and produces many unwanted diffs today. #26187 tracks fixing it.